pydantic/networks.py

```diff
@@ -46,7 +46,7 @@
 _url_regex_cache = None
 _ascii_domain_regex_cache = None
 _int_domain_regex_cache = None
-_domain_ending = r'(?P<tld>\.[a-z]{2,63})?\.?'
+_domain_ending = r'(?P<tld>(\.[^\W\d_]{2,63})|(\.(?:xn--)[_0-9a-z-]{2,63}))?\.?'
```

Comment:
This needs to go into a function so it's only compiled when it's needed. Same as `url_regex`.

Reply:
When `_domain_ending` is defined, it is a raw string; regex compilation only happens within the functions `ascii_domain_regex` and `int_domain_regex`. I think this means the goal of only calling `re.compile` when necessary is already satisfied, without requiring a change.

Is there an additional performance concern? If so, happy to pull this into a function that `ascii_domain_regex` and `int_domain_regex` call.

Reply:
sorry I was being dumb, reviewing on my phone and not looking properly. 

all good :+1: